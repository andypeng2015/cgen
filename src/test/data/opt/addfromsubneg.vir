module foo

export function w $foo(w %x, w %y) {
@start:
  %y = neg.w %y
  %x = sub.w %x, %y
  ret %x
}
