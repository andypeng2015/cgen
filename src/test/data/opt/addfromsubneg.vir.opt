module foo

export function w $foo(w %x, w %y) {
@2:
  %1 = add.w %x, %y ; @6
  ret %1
}
