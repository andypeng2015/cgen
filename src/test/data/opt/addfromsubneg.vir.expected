module foo

export function w $foo(w %x, w %y) {
@0:
  %0 = add.w %x, %y ; @3
  ret %0
}
